hotham/src/rendering/vertex.rs

```diff
@@ -48,11 +48,26 @@ impl Vertex {
             Vector3<f32>,
             Vector4<f32>,
             Vector2<f32>,
-            Vector4<f32>,
+            Vector4<u8>,
             Vector4<f32>,
         ),
     ) -> Self {
-        Vertex::new(t.0, t.1, t.2, t.3, t.4, t.5)
+        let max_weight = t.5.max().max(f32::EPSILON);
```

Comment:
Could you add some comments here just explaining what's going on? I can kind of intuit what you're doing but it would really help to have a more detailed explanation